### PR TITLE
Add rotate control to IIIFImageViewer

### PR DESCRIPTION
**Pull request by davidpmccormick**
Relates to #4442

Adding a rotate (90º clockwise) control to the `IIIFImageViewer`.

Screenshot 2019-05-07 at 17 12 16

### CODE REVIEW

File: common/views/components/ImageViewer/ImageViewer.js [hunk 1]
```diff
@@ -82,6 +82,15 @@ const ViewerContent = ({
   return (
     <div className={`${classes} image-viewer__content image-viewer__content2`}>
       <div className="image-viewer__controls flex flex-end flex--v-center">
+        <Control
+          type="light"
+          text="Rotate"
+          id={`rotate-right-${id}`}
+          icon="rotateRight"
+          extraClasses={`${spacing({ s: 1 }, { margin: ['right'] })}`}
+          clickHandler={handleZoomIn}
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]: `handleZoomIn`?

File: common/views/components/ImageViewer/ImageViewer.js [hunk 2]
```diff
@@ -90,6 +90,9 @@ const ViewerContent = ({
   return (
     <div className={`${classes} image-viewer__content image-viewer__content2`}>
       <div className="image-viewer__controls flex flex-end flex--v-center">
+        {/* Setting OpenSeaDragon's `showRotationControl` option to `true` means we have to provide
+        a dummy `rotateLeftButton` or handle rotation programatically. The former is simpler. */}
+        <span id={`rotate-left-${id}`} className={'is-hidden'} />
```
Comment (davidpmccormick) [inline common/views/components/ImageViewer/ImageViewer.js]: How do we feel about this? To me it felt better than querying for the rotateRight button and adding a click handler to get/set rotation?

Comment (jamesgorrie) [reply common/views/components/ImageViewer/ImageViewer.js]: We could start getting a little more programmatic about it and use the methods exposed on the `Viewer` / `Viewport`?

https://openseadragon.github.io/docs/OpenSeadragon.Viewport.html#setRotation

Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewer.js]: Yeah, that's what I was wondering. Is there a more React-y way of doing this programmatically than e.g.

```
useEffect(() => {
    if (viewer) {
      viewer.destroy();
      setViewer(null);
    }

    setupViewer(infoUrl, id, handleScriptError).then(viewer => {
      const rotateButton = document.getElementById(`rotate-right-${id}`);

      rotateButton &&
        rotateButton.addEventListener('click', () => {
          viewer.viewport.setRotation(viewer.viewport.getRotation() + 90);
        });

      setViewer(viewer);
    });
  }, [infoUrl]);
```

?

Comment (jamesgorrie) [reply common/views/components/ImageViewer/ImageViewer.js]: You could add the event to the button the normal way and have a check on it.
```JSX
{viewer &&
  <Control
    onClick={() => viewer.viewport.setRotation(
      viewer.viewport.getRotation() + 90
    )}} />
```

**Pull request merged**